### PR TITLE
feat(ingestion/okta): okta stateful ingestion

### PR DESCRIPTION
**Pull request by mohdsiddique**
- stateful ingestion support in okta connector 
- Update doc annotation of Azure AD connector

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/identity/okta.py [hunk 1]
```diff
@@ -407,6 +445,14 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
         # Step 4: Close the event loop
         event_loop.close()
 
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        return auto_stale_entity_removal(
+            self.stale_entity_removal_handler,
+            auto_workunit_reporter(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/identity/okta.py:451]: The auto_workunit_reporter is not appropriate here, but the other two helpers are correct

Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/identity/okta.py]: done

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/identity/okta.py [hunk 2]
```diff
@@ -41,11 +52,15 @@
...
+from datahub.utilities.source_helpers import (
+    auto_stale_entity_removal,
+    auto_status_aspect,
+)
 
 logger = logging.getLogger(__name__)
 
 
-class OktaConfig(ConfigModel):
+class OktaConfig(StatefulIngestionConfigBase, DatasetSourceConfigMixin):
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/identity/okta.py:63]:
```suggestion
class OktaConfig(StatefulIngestionConfigBase):
```

**Review by hsheth2 — CHANGES_REQUESTED** [review]
one minor change

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
@mohdsiddique looks like lint is failing

**Pull request merged**